src/Mod/Path/libarea/Adaptive.cpp

```diff
@@ -146,6 +146,38 @@ inline DoublePoint GetPathDirectionV(const Path &pth, size_t pointIndex)
 	const IntPoint &p2 = pth.at(pointIndex);
 	return DirectionV(p1, p2);
 }
+	
+bool doFilter(Path &ppg) {
+        size_t i;
+        size_t b;
+        size_t start;
+        size_t end;
+        for(i=0;i<ppg.size();i++) {
+            if (i==ppg.size()-1) {
+                start=0;
+                end=ppg.size()-2;
+            } else {
+                start=i+1;
+                end=ppg.size();
+            }
+            for(b=start;b<end;b++) {
+                if (abs(ppg[i].X-ppg[b].X)<2 && abs(ppg[i].Y-ppg[b].Y)<2) {
+                    ppg.erase(ppg.begin()+b);
+                    return true;
+                }
+            }
+        }
+        return false;
+}
+
+void filterCloseValues(Paths &ppg) {
+        size_t i;
+        for(i=0;i<ppg.size();i++) {
```

Comment:
In new code it's best to use a range-based for loop.

Reply:
hmm I need that position information and also the next position index; how would you do that?